atom/browser/atom_browser_main_parts.cc

```diff
@@ -414,6 +438,27 @@ void AtomBrowserMainParts::PreMainMessageLoopRun() {
   node_bindings_->PrepareMessageLoop();
   node_bindings_->RunMessageLoop();
 
+#if BUILDFLAG(ENABLE_ELECTRON_EXTENSIONS)
+  extensions_client_ = std::make_unique<AtomExtensionsClient>();
+  extensions::ExtensionsClient::Set(extensions_client_.get());
+
+  // BrowserContextKeyedAPIServiceFactories require an ExtensionsBrowserClient.
+  extensions_browser_client_ = std::make_unique<AtomExtensionsBrowserClient>();
+  extensions::ExtensionsBrowserClient::Set(extensions_browser_client_.get());
+
+  extensions::EnsureBrowserContextKeyedServiceFactoriesBuilt();
+  extensions::atom::EnsureBrowserContextKeyedServiceFactoriesBuilt();
+
+  browser_context_ = AtomBrowserContext::From("", false);
```

Comment:
The extensions system is closely tied to a `BrowserContext`. In the case of Electron, a [`Session`](https://electronjs.org/docs/api/session#class-session) maps directly to a `BrowserContext` instance.

Under the followup roadmap in this PR, the first entry is `Add support for extensions in multiple BrowserContexts`. I think this will cover moving the extensions initialization logic into the `AtomBrowserContext` class.